packages/vite/src/node/optimizer/scan.ts

```diff
@@ -143,7 +143,7 @@ function globEntries(pattern: string | string[], config: ResolvedConfig) {
 const scriptModuleRE =
   /(<script\b[^>]*type\s*=\s*(?:"module"|'module')[^>]*>)(.*?)<\/script>/gims
 export const scriptRE = /(<script\b(?:\s[^>]*>|>))(.*?)<\/script>/gims
-export const commentRE = /<!--(.|[\r\n])*?-->/
+export const commentRE = /<!--(.|[\r\n])*?-->/g
```

Comment:
👍  [demo](https://regex101.com/r/LlfplX/1)

I'm wondering if a new test should be added [here](https://github.com/vitejs/vite/blob/bf0b631e7479ed70d02b98b780cf7e4b02d0344b/packages/vite/src/node/__tests__/scan.spec.ts#L29) to verify support for multiple comments.

Reply:
Yep, I second adding a test there for this too. I think for now using the `g` flag only should be alright.

Reply:
thanks for pointing me to the test. I've updated the existing test to have two comments. I added the `s` flag as well, which I agree is cleaner